uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Form/Search/StaffDirectory.php

```diff
@@ -0,0 +1,390 @@
+   *
+   * @param string $jobDetailsCondition
+   *
+   * @return string
+   */
+  private function getJobContractJoin($jobDetailsCondition) {
+    $jobDetailsQuery =  "SELECT * FROM civicrm_hrjobcontract_details contract_details ";
+    if ($jobDetailsCondition) {
+      $jobDetailsQuery .= "WHERE {$jobDetailsCondition}";
+    }
+
+    $jobDetailsQuery .= " ORDER BY period_start_date DESC";
+
+    $sql = "LEFT JOIN (SELECT civicrm_hrjobcontract_details.title, civicrm_hrjobcontract.id, civicrm_hrjobcontract.contact_id,
+      civicrm_hrjobcontract_details.period_start_date, civicrm_hrjobcontract_details.period_end_date
+      FROM civicrm_hrjobcontract 
+      INNER JOIN civicrm_hrjobcontract_revision rev 
+      ON rev.id = (SELECT id
+                   FROM civicrm_hrjobcontract_revision jcr2
+                   WHERE jcr2.jobcontract_id = civicrm_hrjobcontract.id
+                   ORDER BY jcr2.effective_date DESC LIMIT 1)
+      INNER JOIN ({$jobDetailsQuery}) civicrm_hrjobcontract_details
+        ON rev.details_revision_id = civicrm_hrjobcontract_details.jobcontract_revision_id 
+      WHERE civicrm_hrjobcontract.deleted = 0 
+      GROUP BY civicrm_hrjobcontract.contact_id) contract_details
+      ON contract_details.contact_id = contact_a.id";
+
+    return $sql;
+  }
+
+  /**
+   * Returns the SQL query condition for the period start and end dates
+   * depending on the staff type.
+   *
+   * @param string $staffType
+   *
+   * @return string
+   */
+  private function getJobDetailsConditionForSpecificStaff($staffType = 'all') {
+    $today = date('Y-m-d');
+    $sql = '';
+
+    if ($staffType == 'current') {
+      $sql .= "contract_details.period_start_date <= '{$today}'
+               AND (contract_details.period_end_date >= '{$today}' OR contract_details.period_end_date IS NULL)";
+    }
+
+    if ($staffType == 'past') {
+      $sql .= "contract_details.period_end_date < '{$today}'";
+    }
+
+    if ($staffType == 'future') {
+      $sql .= "contract_details.period_start_date > '{$today}'";
+    }
+
+    return $sql;
+  }
+
+  /**
+   * Returns the SQL query condition for the period start and end dates
+   * depending on the given period start and end dates.
+   *
+   * @param \DateTime|NULL $periodStartDate
+   * @param \DateTime|NULL $periodEndDate
+   *
+   * @return string
+   */
+  private function getJobDetailsConditionForSpecificDates(
+    DateTime $periodStartDate = NULL,
+    DateTime $periodEndDate = NULL)
+  {
+
+    $sql = '';
+    if (!$periodStartDate && !$periodEndDate) {
+      return $this->getJobDetailsConditionForSpecificStaff();
+    }
+
+    if ($periodStartDate) {
+      $sql .= "(contract_details.period_end_date >= '" .
+        $periodStartDate->format('Y-m-d') . "' OR contract_details.period_end_date IS NULL) AND ";
+    }
+
+    if ($periodEndDate) {
+      $sql .= "contract_details.period_start_date <= '". $periodEndDate->format('Y-m-d') . "' AND ";
+    }
+
+    return rtrim($sql, 'AND ');
```

Comment:
In this case, can't we add the conditions to an array and then join them with an `implode(' AND ', $conditions)` at the end? I think this will be clearer than the `rtrim()`